source/common/network/io_socket_handle_impl.cc

```diff
@@ -304,10 +304,13 @@ Api::IoCallUint64Result IoSocketHandleImpl::recvmsg(Buffer::RawSlice* slices,
     return sysCallResultToIoCallResult(result);
   }
 
-  RELEASE_ASSERT((hdr.msg_flags & MSG_CTRUNC) == 0,
-                 fmt::format("Incorrectly set control message length: {}", hdr.msg_controllen));
-  RELEASE_ASSERT(hdr.msg_namelen > 0,
-                 fmt::format("Unable to get remote address from recvmsg() for fd: {}", fd_));
+  if ((hdr.msg_flags & MSG_CTRUNC) != 0 || hdr.msg_namelen <= 0) {
+    ENVOY_BUG((hdr.msg_flags & MSG_CTRUNC) == 0,
+              fmt::format("Incorrectly set control message length: {}", hdr.msg_controllen));
+    ENVOY_BUG(hdr.msg_namelen > 0,
+              fmt::format("Unable to get remote address from recvmsg() for fd: {}", fd_));
+    return sysCallResultToIoCallResult(result);
```

Comment:
I think this will indicate success to the caller but have no peer address. I'm unsure if it will cause the upper layer to misbehave in some weird way. It looks like it may better to treat this case the same as the data truncation above and set the truncated_and_dropped_ flag and dropped_packets_ flags.